templates/s3/s3.py

```diff
@@ -114,22 +114,36 @@ def _ensure_namespaces_connections(self):
             self.data['current_namespaces_connections'] = namespaces_connection
 
         self.logger.info("verify tlog namespace connections")
-        if 'tlog' not in self.data or not self.data['tlog'].get('node') or not self.data['tlog'].get('node'):
-            return
-
-        robot = get_zrobot(self.data['tlog']['node'], self.data['tlog']['url'])
-        namespace = robot.services.get(template_uid=NS_TEMPLATE_UID, name=self.data['tlog']['name'])
-
-        connection_info = namespace_connection_info(namespace)
-        if self.data['tlog'].get('address') and self.data['tlog']['address'] == connection_info:
-            self.logger.info("tlog namespace connection in service data is in sync with reality")
-            return
+        tlog = self.data.get('tlog', {})
```

Comment:
`_ensure_namespaces_connections` become to be to be pretty big, we should split it into different functions